### PR TITLE
Adding support for providing Auth headers via Swagger UI

### PR DESCRIPTION
**Pull request by ayush3797**
## Why make this change?
To enable users to provide value for `X-MS-API-ROLE` and `Authorization` headers via Swagger UI. Addresses: https://github.com/Azure/data-api-builder/issues/1789

## What is this change?
Added method `OpenApiDocumentor.AddCustomHeadersToOperation` which will add the above mentioned two headers for every operation supported for every entity irrespective of the entity type (Table/View/StoredProcedure).

## How was this tested?
- [x] Unit Tests - Added via `ParameterValidationTests.ValidateHeaderParametersForEntity`.

### CODE REVIEW

**Review by seantleonard — COMMENTED** [review]
looks good so far with surfacing the authN/Z headers. In my opinion, any enablement of auth flow usage can come in a separate PR to keep this PR small and intentional.

File: src/Core/Services/OpenAPI/OpenApiDocumentor.cs [hunk 1]
```diff
@@ -435,20 +450,52 @@ private OpenApiOperation CreateBaseOperation(string description, List<OpenApiTag
...
+                Schema = stringParamSchema
+            };
+            operation.Parameters.Add(paramForClientHeader);
+
+            // Add parameter for Authorization header.
+            OpenApiParameter paramForAuthHeader = new()
+            {
+                Required = false,
+                In = ParameterLocation.Header,
+                Name = "Authorization",
```
Comment (Aniruddh25) [inline src/Core/Services/OpenAPI/OpenApiDocumentor.cs:484]: Can we create a constant for this header just like `CLIENT_ROLE_HEADER`?

File: src/Service.Tests/OpenApiDocumentor/ParameterValidationTests.cs [hunk 2]
```diff
@@ -236,4 +238,46 @@ private static void AssertOnAllDefaultQueryParameters(List<OpenApiParameter> ope
...
+    [DataTestMethod]
+    [DataRow("BooksTable", "books", EntitySourceType.Table, DisplayName = "Assert custom header presence in header parameters for table.")]
+    [DataRow("BooksView", "books_view_all", EntitySourceType.View, DisplayName = "Assert custom header presence in header parameters for view.")]
+    [DataRow("UpdateBookTitle", "update_book_title", EntitySourceType.StoredProcedure, DisplayName = "Assert custom header presence in header parameters for stored procedure.")]
+    [DataRow("BooksTable", "books", EntitySourceType.Table, DisplayName = "Validate custom header presence in header parameters for table.")]
+    [DataRow("BooksView", "books_view_all", EntitySourceType.View, DisplayName = "Validate custom header presence in header parameters for view.")]
+    [DataRow("UpdateBookTitle", "update_book_title", EntitySourceType.StoredProcedure, DisplayName = "Validate custom header presence in header parameters for stored procedure.")]
+    public async Task ValidateHeaderParametersForEntity(string entityName, string objectName, EntitySourceType entitySourceType)
+    {
+        EntitySource entitySource = new(Object: objectName, entitySourceType, null, null);
```
Comment (Aniruddh25) [inline src/Service.Tests/OpenApiDocumentor/ParameterValidationTests.cs:259]: nit: name the arguments that are null here

File: src/Service.Tests/OpenApiDocumentor/ParameterValidationTests.cs [hunk 3]
```diff
@@ -236,4 +238,46 @@ private static void AssertOnAllDefaultQueryParameters(List<OpenApiParameter> ope
...
+        OpenApiDocument openApiDocument = await GenerateOpenApiDocumentForGivenEntityAsync(entityName, entitySource);
+        foreach (OpenApiPathItem pathItem in openApiDocument.Paths.Values)
+        {
+            foreach ((OperationType operationType, OpenApiOperation operation) in pathItem.Operations)
+            {
+                Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Header && "Authorization".Equals(param.Name) && JsonDataType.String.ToString().ToLower().Equals(param.Schema.Type)));
+                Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Header && AuthorizationResolver.CLIENT_ROLE_HEADER.Equals(param.Name) && JsonDataType.String.ToString().ToLower().Equals(param.Schema.Type)));
+                // Assert presence of Authorization header and the expected parameter properties for the header parameter.
+                Assert.IsTrue(operation.Parameters.Any(
```
Comment (Aniruddh25) [inline src/Service.Tests/OpenApiDocumentor/ParameterValidationTests.cs:268]: Isn't this assertion providing a superset of the same assertion as on line 265? Same for the assertion on CLIENT_ROLE_HEADER

**Review by Aniruddh25 — APPROVED** [review]
Looks good, left few nits

File: src/Service.Tests/OpenApiDocumentor/ParameterValidationTests.cs (hunk 3 above)
Comment (ayush3797) [reply src/Service.Tests/OpenApiDocumentor/ParameterValidationTests.cs:268]: Oops yeah!

**Review by Aniruddh25 — APPROVED** [review]
LGTM!

File: src/Core/Services/OpenAPI/OpenApiDocumentor.cs [hunk 4]
```diff
@@ -344,6 +347,18 @@ private OpenApiPaths BuildPaths()
             }
         }
 
+        /// <summary>
+        /// Helper method to add query parameters like $select, $first, $orderby etc. to get,getAll operations for tables/views.
```
Comment (aaronburtle) [inline src/Core/Services/OpenAPI/OpenApiDocumentor.cs:351]:
```suggestion
        /// Helper method to add query parameters like $select, $first, $orderby etc. to get and getAll operations for tables/views.
```

File: src/Core/Services/OpenAPI/OpenApiDocumentor.cs [hunk 5]
```diff
@@ -435,20 +450,52 @@ private OpenApiOperation CreateBaseOperation(string description, List<OpenApiTag
...
+                Required = false,
+                In = ParameterLocation.Header,
+                Name = "Authorization",
+                Schema = stringParamSchema
+            };
+            operation.Parameters.Add(paramForAuthHeader);
+        }
+
         /// <summary>
         /// This method creates a list of OpenApiParameter objects for the input parameters of a stored procedure.
```
Comment (aaronburtle) [inline src/Core/Services/OpenAPI/OpenApiDocumentor.cs:491]:
```suggestion
        /// This method adds the input parameters from the stored procedure definition to the OpenApi operation parameters.
```

**Review by aaronburtle — APPROVED** [review]
Looks good, just a couple of nits/suggestions for updating comments.

**Pull request merged**